File: 0383-ransom-note/0383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        target, source = map(sorted, (ransomNote, magazine))
        m, n = map(len, (target, source))
        i = j = 0
        while True:
            if i == m:
                return True
            if j == n:
                return False
            if target[i] == source[j]:
                i += 1
            j += 1

File: 0383-ransom-note/0383-ransom-note.ts
function canConstruct(ransomNote: string, magazine: string): boolean {
    const [ransom_cnt, magazine_cnt] = Array.from([ransomNote, magazine], (e) => (
        e.split('').reduce((acc, curr) => {
            acc[curr] ??= 0
            acc[curr] += 1
            return acc
        }, {})
    ))
    return Object
        .keys(ransom_cnt)
        .every((key) => ransom_cnt[key] <= (magazine_cnt[key] ?? 0))
};
